pkgs/by-name/an/ananicy-cpp/match-wrappers.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/src/worker.cpp b/src/worker.cpp
+index 0cf8955..0c142b4 100644
+--- a/src/worker.cpp
++++ b/src/worker.cpp
+@@ -6,6 +6,7 @@
+ #include "core/cgroups.hpp"
+ #include "core/priority.hpp"
+ #include "utility/process_info.hpp"
++#include "utility/regex_utils.hpp"
+ #include "version.hpp"
+ 
+ #include <chrono>
+@@ -29,7 +30,10 @@ void Worker::work(const std::stop_token &stop_token) {
+   while (!stop_token.stop_requested()) {
+     while ((proc = process_queue->poll(500ms)).has_value()) {
+       const auto &p = proc.value();
+-      const auto &rule = rules->get_rule(p.name);
++      auto name = p.name;
++      if (utils::regex::is_str_matching_regex(name, "^\\..+-wrap(p(ed?)?)?$"))
++        name = name.substr(1, name.find_last_of('-') - 1);
```

Comment:
So it matches -wrapped and then removes the -wrapped right?

What's up with the `(p(ed?)?)`?

Reply:
AFAIK, sometimes process names don't fully have "-wrapped" suffix in their name, due to size limitations.

Reply:
It matches
```
.x-wrap
.x-wrapp
.x-wrappe
.x-wrapped
```
and removes everything after the last `-` (the `-` included).

Reply:
Not all process names end like that, as you can see. 

https://github.com/CachyOS/ananicy-rules/blob/e3638ebb2320202ede9facfc9657cc8af2098a2d/00-default/Services/gvfs.rules

![image](https://github.com/NixOS/nixpkgs/assets/50095635/588edb16-2bf9-4e63-9b15-e8ed8c874948)


I had to manually add these rules to match `.x-` `.x-w`, `.x-wra` and so on.

All wrapped app's processes have `.` in the beginning. 
I think the rules should run on any process `x` and `\.x.*` (regex)

Reply:
@JohnRTitor are you sure that screenshots includes the full process names?

Reply:
Looks like they do, in which case I'll try to come up with a better way of stripping the extra characters. I'm not sure how yet, so ideas are welcome.

Reply:
Never mind, it looks like `ps` is doing something funky. If you check the [nixpkgs makeWrapper implementation](https://github.com/NixOS/nixpkgs/blob/683aa7c4e385509ca651d49eeb35e58c7a1baad6/pkgs/build-support/setup-hooks/make-wrapper.sh#L219), you can see that wrapers always start with a `.` and end with `-wrapped`. You can also confirm this by check the store paths for programs, eg.: `ls -a $(dirname $(realpath $(which firefox)))`. And fortunately the process names in ananicy-cpp are also correct, why I confirmed by logging:

![tmp ElXKWq8NkR](https://github.com/NixOS/nixpkgs/assets/55629891/700de37c-b316-44cd-bea6-b7b9e4b44506)

(The right side is `ps`, the left side is a log line from ananicy-cpp)